src/ebonite/build/helpers.py

```diff
@@ -1,12 +1,19 @@
 from functools import wraps
-from typing import List
+from typing import List, Callable
 from pyjackson.utils import get_function_fields, turn_args_to_kwargs
 
 
-def validate_kwargs(f=None, *, allowed: List[str] = None):
+def validate_kwargs(f=None, *, allowed: List[str] = None, allowed_funcs: List[Callable] = None):
 
     def inner(func):
-        all_fields = set(allowed or [])
+        all_fields = allowed or []
+        if allowed_funcs is not None:
+            from_funcs = []
+            [from_funcs.extend(get_function_fields(allowed_func, False)) for allowed_func in allowed_funcs]
```

Comment:
why not `ff = [f for func in funcs for f in get_fields(func)]`?

Reply:
i considered it to be less readable

Reply:
for loop is better for readability (but you already have pretty nested code here). list comprehensions with side effects are evil in any case

Reply:
Got you.